### PR TITLE
Add Signal.app v1.0.34

### PR DESCRIPTION
**Pull request by kusold**
Homepage: https://signal.org/download/
Github Repo: https://github.com/WhisperSystems/Signal-Desktop

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by alecglassford** [comment]
Since Signal Desktop is published via GitHub Releases, maybe you could add an [appcast](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md)? I guess it would be `https://github.com/WhisperSystems/Signal-Desktop/releases.atom` - though I do see that the current release is actually marked as "pre-release"; I don't know if that would muck with things. I'm guessing this is an artifact of the transition from Chrome and that future releases wouldn't have the pre-release flag (cc: @scottnonnenberg).

Also, maybe you could use `version` in the URL, like [this](https://github.com/caskroom/homebrew-cask/pull/40202/files#diff-e89083433a404322f978caf2dbfcb6f2R5), just for simplifying future updates.

So excited for this!

**Comment by kusold** [comment]
@alecglassford Updated the url to read the version.

File: Casks/signal.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'signal' do
+  version '1.0.34'
+  sha256 'b45f89aeb9efd7d28eba8b2fdc5dd2fb0ca090daca3641afb1b2a549988337b3'
+
+  url "https://updates.signal.org/desktop/Signal-mac-#{version}.zip"
+  name 'Signal'
+  homepage 'https://signal.org/'
+
+  app 'Signal.app'
+end
```
Comment (0xadada) [inline Casks/signal.rb]: Could you add a `zap` stanza?

```ruby
  zap delete: '~/Library/Saved Application State/org.whispersystems.signal-desktop.savedState',
      trash:  [
                '~/Library/Application Support/Signal',
                '~/Library/Preferences/org.whispersystems.signal-desktop.helper.plist',
                '~/Library/Preferences/org.whispersystems.signal-desktop.plist',
              ]
```

**Review by 0xadada — CHANGES_REQUESTED** [review]

File: Casks/signal.rb (hunk 1 above)
Comment (kusold) [reply Casks/signal.rb]: Updated!

**Review by 0xadada — APPROVED** [review]

**Comment by oncletom** [comment]
If I understand well the `appcast` thingy, the following stanza could be added:

```
  appcast 'https://github.com/WhisperSystems/Signal-Desktop/releases.atom',
        checkpoint: 'a13cfed5bfed095498049f8a8ca83ac58326b907f7b58defd561defdaf5710d8'
```

Is there a command which checks if a new release has been made?

**Comment by scottnonnenberg** [comment]
I just marked Signal Desktop `v1.0.34` as _not_ a prerelease. We have beta builds listed on there, so it is important to pay attention to that flag. :0)

**Review by commitay — APPROVED** [review]

**Pull request merged**